app/controllers/admin/staff_action_logs_controller.rb

```diff
@@ -9,7 +9,14 @@ def index
     page = (params[:page] || 0).to_i
     page_size = fetch_limit_from_params(default: INDEX_LIMIT, max: INDEX_LIMIT)
 
-    staff_action_logs = UserHistory.staff_action_records(current_user, filters)
+    start_date = params[:start_date]&.to_time
+    end_date = params[:end_date]&.to_time
+
+    query = UserHistory
+    query = query.where("created_at >= ?", start_date) if start_date.present?
+    query = query.where("created_at <= ?", end_date) if end_date.present?
+
+    staff_action_logs = query.staff_action_records(current_user, filters)
```

Comment:
Hmm. this looks a bit weird.

`staff_action_records` takes in options, and I feel `opts` within that `staff_action_records` method iself should include the `start_date` and `end_date` rather than us making the query here. Should we move this query into that method instead?

Reply:
```ruby
    staff_action_logs = UserHistory.staff_action_records(current_user, filters)
    staff_action_logs = staff_action_logs.where("created_at >= ?", start_date) if start_date.present?
    staff_action_logs = staff_action_logs.where("created_at <= ?", end_date) if end_date.present?
```

how about this?

Reply:
Oh, I meant `filters={}` to include `start_date` and `end_date`, so there isn't a need to have another query in the controller itself. The query is done in the `staff_action_records` method

Reply:
Actually, I think this violates the principle of least responsibility. But after checking the code elsewhere, I have to admit that what you said makes sense.

Reply:
The way I see it is that start and end dates are filters, and controllers should ideally not be making model queries (ideally at a lower level).

Thanks for the change 👌🏻